### PR TITLE
fix delete to start at cluster mgr and replicate through each node

### PR DESCRIPTION
**Pull request by tsloughter**
This cleans up a few things and fixes the test for delete. Opening a corresponding PR on nucleus that will use this now when a channel or device is deleted.

### CODE REVIEW

File: src/vg_client.erl [hunk 1]
```diff
@@ -180,10 +181,22 @@ replicate(Pool, Topic, ExpectedId, RecordBatch, Timeout) ->
...
+                {ok, ok} -> ok;
+                {error, Reason} -> {error, Reason}
+            end;
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
 delete_topic(Pool, Topic) ->
-    lager:debug("delete_topic pool=~p topic=~p", [Pool, Topic]),
+            lager:debug("delete_topic pool=~p topic=~p", [Pool, Topic]),
```
Comment (varnerac) [inline src/vg_client.erl]: Whacky indentation

File: src/vg_conn.erl [hunk 2]
```diff
@@ -266,12 +266,29 @@ handle_request(?REPLICATE_REQUEST, _Role, Data, CorrelationId, Socket) ->
...
     lager:info("received request to delete topic ~p", [Topic]),
-    Resp = case vg_topic_mgr:delete_topic(Topic, 0) of
-               ok -> <<"OK">>;
-               {error, Reason} ->
-                   io_lib:format("~p", [Reason])
-           end,
-    Response = vg_protocol:encode_string(Resp),
+    try vg:delete_topic(Topic) of
+        ok -> ok;
+        {error, Reason} ->
```
Comment (varnerac) [inline src/vg_conn.erl]: Put both try cases on one line or split them both out.

File: src/vg_conn.erl [hunk 3]
```diff
@@ -266,12 +266,29 @@ handle_request(?REPLICATE_REQUEST, _Role, Data, CorrelationId, Socket) ->
...
+    Response = vg_protocol:encode_string(<<"OK">>),
+    Size = erlang:iolist_size(Response) + 4,
+    gen_tcp:send(Socket, [<<Size:32/signed-integer, CorrelationId:32/signed-integer>>, Response]);
+handle_request(?REPLICATE_DELETE_TOPIC_REQUEST, _Role, Data, CorrelationId, Socket) ->
+    {Topic, <<>>} = vg_protocol:decode_string(Data),
+    lager:info("received replicated request to delete topic ~p", [Topic]),
+    try vg_topic_mgr:delete_topic(Topic, 0) of
+        ok -> ok;
+        {error, Reason} ->
+            lager:info("at=delete_topic error=~p", [Reason])
```
Comment (varnerac) [inline src/vg_conn.erl]: All on one line or all split out

File: test/z_cluster_SUITE.erl [hunk 4]
```diff
@@ -560,9 +560,16 @@ deactivate_list_topics(Config) ->
      end
      || N <- lists:seq(1, 10)],
 
+    [[?assert(filelib:is_dir(filename:join("data/node" ++ Node, <<"quux", (integer_to_binary(N))/binary, "-0">>)))
+     || N <- lists:seq(5, 14)] || Node <- ["0", "1", "2"]],
+
     %% delete a subset including some that are not active
     [ok = rpc:call(Head, vg, delete_topic, [<<"quux", (integer_to_binary(N))/binary>>])
-     || N <- lists:seq(5, 14)],
+        || N <- lists:seq(5, 14)],
```
Comment (varnerac) [inline test/z_cluster_SUITE.erl]: Indentation inconsistent with others

**Review by varnerac — CHANGES_REQUESTED** [review]

File: src/vg_client.erl (hunk 1 above)
Comment (tsloughter) [reply src/vg_client.erl]: I'm a wacky guy.

Comment (varnerac) [reply src/vg_client.erl]: Truth

**Comment by tsloughter** [comment]
@varnerac you good with the changes I've made?

**Review by varnerac — APPROVED** [review]

**Pull request merged**